lib/smack/SmackRep.cpp

```diff
@@ -154,8 +210,8 @@ void SmackRep::collectRegions(llvm::Module &M) {
 
 const Stmt* SmackRep::alloca(llvm::AllocaInst& i) {  
   const Expr* size = 
-    Expr::fn("$mul",lit(storageSize(i.getAllocatedType())),lit(i.getArraySize()));
-                       
+    Expr::fn(opName("$mul", {TRUNC(getIntSize(i.getOperand(0)))}), lit(storageSize(i.getAllocatedType())),lit(i.getArraySize()));
```

Comment:
I would say that the two arguments of alloca has little to do with pointer arithmetic. And the size of two arguments (the size of type allocated and the number of this type variables) of "$mul" is usually 32 because it is rare for a single type to take more than 4 GB.